### DISCUSSION TITLE
Simple question about filter syntax

### DISCUSSION THREAD

**Discussion by Tiagoquix** [uBlockOrigin/uBlock-issues#2998] [Q&A] [answered]
Is there any difference between the following filters?

```
||example.com/subpage/^$removeparam=test
```

```
||example.com/subpage/*^$removeparam=test
```

(The last one has a wildcard.)

**Comment by krystian3w** [comment]
You probably only need to understand what the `^` at the end of the filters with the 'directory' path does.

it is easier to understand how the `|` at the end of the filter works (almost similar to `$` in regex).

**Comment by ghost** [comment, marked as answer]
https://help.adblockplus.org/hc/en-us/articles/360062733293#separators

**Comment by Yuki2718** [comment]
As already said, adding the separator here only makes the rule unworkable.  If no such separator, no difference as ABP synatx basically assumes wildcard before and after each rule (in this case not before the rule though, thx to domain anchor). Read the documents carefully first.
